RIOT-OS/RIOT
# sys/ztimer: add modules to Kconfig

### Contribution description
This PR adds modelling of the `ztimer%` modules (at least all the ones that I've found). Dependencies and defaults in the current Makefile.dep are quite complex, so this needs some careful reviewing. This also includes some fixes in the current xtimer Kconfig dependencies. All xtimer and ztimer tests include now their configuration files and are added to the migration tests.

**Note**: #15715 is introducing a new module so this will probably need some adaption if that gets merged before.

There are some modules which I was not really sure if the user should be able to disable (e.g. `ztimer_core`), so you will find that they have no prompt. Some feedback on this would be appreciated, as I'm not really familiar with the module.

This PR is needed by another one that I have in my queue, continuing the series of PR modelling drivers in Kconfig.

### Testing procedure
- Check that the dependencies are correctly modelled
- Check that the new symbols are only visible and usable when TEST_KCONFIG=1
- Check that the correspondent applications for the modelled drivers were added to the CI's checks
- Green CI

### Issues/PRs references
- #15715
